core/market-decentralized/examples/discovery_usage.rs

```diff
@@ -0,0 +1,97 @@
+use anyhow;
+use async_trait::async_trait;
+use chrono::Utc;
+use serde_json::json;
+use std::sync::Arc;
+use tokio::runtime::Runtime;
+
+use ya_client_model::market::Offer;
+use ya_market_decentralized::protocol::callbacks::HandlerSlot;
+use ya_market_decentralized::protocol::{
+    Discovery, DiscoveryBuilder, DiscoveryError, DiscoveryFactory, DiscoveryInitError,
+    OfferReceived, RetrieveOffers,
+};
+
+// =========================================== //
+// TODO: Remove this example after implementing Discovery
+// =========================================== //
```

Comment:
I wanted to avoid leaving this... This was just a test. Discovery will be reference implementation.
We will have to maintain this code and t will be unnecessary pain.